### ISSUE TITLE
Setprop only works with strings

### ISSUE THREAD

**Issue by elFua** [bjorns/aem-cmd#19]
Setting booleans or arrays does not work as expected.

Steps:
---

    acmd setprop acmd1="{Boolean}true",acmd2="[1,2,3,4,5]" /content/geometrixx/jcr:content

Expected:
---
A boolean and an array.

Actual:
---
Two string properties:

    acmd1={Boolean}true
    acmd2=[1

**Comment by bjorns** [comment]
Well, technically no other expectation has been set than strings but yes, I agree it would be nice to be able to write other types.

Do you have a link to how this works if I use curl?

**Comment by elFua** [comment]
@bjorns: In fact I realize now this is not a bug, but rather a feature request, because users can set the type via `TypeHint`'s.

The `curl` syntax would be:
```bash
     curl -vv -u admin:admin -X POST --data 'aemFlag=true&aemFlag@TypeHint=Boolean&aemArr=[1,2,3,4,5]&aemArr@TypeHint=String[]' 
```
or the same thing but in a more readable form:
```bash
    curl -vv -u admin:admin  \
        -F aemFlag=true \
        -F aemFlag@TypeHint="Boolean" \
        -F aemArr="[1,2,3,4]" \
        -F "aemArr@TypeHint=String[]" \
        http://cq:6200/etc/tags/workflow-package/rollout-en \
```
Adding support for it in the tool would mean interpreting value prefixes like `{Boolean}true` and converting them to extra TypeHint parameters, as documented at https://sling.apache.org/documentation/bundles/manipulating-content-the-slingpostservlet-servlets-post.html#typehint

For booleans that are set to false that would also involve adding the `@Delete:true` type hint, since CQ deletes false properties rather than setting them to {Boolean}false.

**Comment by bjorns** [comment]
Ok, I think I'm going to put some basic type recognition and add a --typeHint flag just in case.

**Comment by bjorns** [comment]
Implemented integer and boolean support in 983666d6eb9e032ea934c4687c2ed4fc60b79bc1

Unfortunately requests doesn't support having multiple form fields with the same name which seems to be what is required to set array data so that will remain known issue for the forseable future.q

**Comment by elFua** [comment]
@bjorns: Just for reference: in fact, requests _can_ send multiple form fields with the same name. This is accomplished by sending a `multipart/form-data` POST request rather than a `application/x-www-form-urlencoded` POST. 

AEM itself uses multipart form encoding for most POST operations rather than `www-form-urlencoded`.

The `requests` equivalent would be `requests.post('http://httpbin.org/post', files=dict(foo='bar'))` which I have found to be tricky to use since it requires ugly and un-intuitive syntax when you don't want to include files, given that `files` is mainly meant for file uploading and the API for using it for multipart form posting without files is imo faulty.

A nicer alternative to using `files` is the `requests_toolbelt` which is an extension for `requests`. I included an example at http://stackoverflow.com/a/35974071 which basically looks like:

```python
    import requests
    from requests_toolbelt.multipart.encoder import MultipartEncoder
    
    multipart_data = MultipartEncoder(
        fields=(
                ('action', 'store'), 
                ('path', '/path1'),
                ('path', '/path2'),
                ('path', '/path3'),
               )
        )

    response = requests.post('http://httpbin.org/post', data=multipart_data, headers={'Content-Type': multipart_data.content_type})

```
